File: leetcode.py/二叉树的直径.py
class Solution(object):
    def diameterOfBinaryTree(self, root):
        self.max_diameter = 0
        def depth(node):
            if not node:
                return 0
            
            left_depth = depth(node.left)
            rigth_depth = depth(node.right)

            current_diameter = left_depth +rigth_depth
            self.max_diameter = max(self.max_diameter,current_diameter)

            return max(left_depth,rigth_depth) + 1
        depth(root)
        return self.max_diameter
# 以下为示例代码，展示如何使用 Solution 类计算二叉树的直径
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

if __name__ == "__main__":
    # 构建一个示例二叉树
    #       1
    #      / \
    #     2   3
    #    / \
    #   4   5
    root = TreeNode(1)
    root.left = TreeNode(2)
    root.right = TreeNode(3)
    root.left.left = TreeNode(4)
    root.left.right = TreeNode(5)

    solution = Solution()
    diameter = solution.diameterOfBinaryTree(root)
    print(f"该二叉树的直径是: {diameter}")  # 预期输出: 3

File: leetcode.js/二叉树的直径.js
var diameterOfBinaryTree = function(root) {
    let max_diameter = 0;
    function depth(node){
        if(!node)return 0;
        let left_depth = depth(node.left);
        let right_depth = depth(node.right);
        let current_diameter = left_depth + right_depth;
        max_diameter = Math.max(max_diameter,current_diameter);
        return Math.max(left_depth,right_depth)+1
    }
    depth(root);
    return max_diameter
};
// 示例使用
// 定义二叉树节点类
class TreeNode {
    constructor(val = 0, left = null, right = null) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

// 构建示例二叉树
//       1
//      / \
//     2   3
//    / \
//   4   5
const root = new TreeNode(1);
root.left = new TreeNode(2);
root.right = new TreeNode(3);
root.left.left = new TreeNode(4);
root.left.right = new TreeNode(5);

// 调用函数计算二叉树的直径
const diameter = diameterOfBinaryTree(root);
console.log("该二叉树的直径是: ", diameter); // 输出: 该二叉树的直径是:  3
